.github/workflows/main.yml

```diff
@@ -0,0 +1,196 @@
+name: Main
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  NODE_VERSION: 18
+  SOLANA_VERSION: 1.18.12
+  CARGO_CACHE: |
+    ~/.cargo/bin/
+    ~/.cargo/registry/index/
+    ~/.cargo/registry/cache/
+    ~/.cargo/git/db/
+    target/
+
+jobs:
+  build_programs:
+    name: Build programs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v4
+      - name: Setup environment
+        uses: ./.github/actions/setup
+        with:
+          node: ${{ env.NODE_VERSION }}
+          solana: ${{ env.SOLANA_VERSION }}
+      - name: Cache cargo dependencies
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.CARGO_CACHE }}
+          key: ${{ runner.os }}-cargo-programs-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-programs
+      - name: Build programs
+        run: pnpm programs:build --features bpf-entrypoint
```

Comment:
Do we need to add the `--features bpf-entrypoint` to the template?

Reply:
I would lean towards no, only because this is likely a pattern we're going to use but not everyone will. So, to me, it makes sense for it to be a second-step configuration we add.

Alternatively, maybe `create-solana-program` can accept an argument for `--program-features` where you specify at template generation time?